mono/metadata/threads.c

```diff
@@ -1259,7 +1259,7 @@ start_wrapper_internal (StartInfo *start_info, gsize *stack_ptr)
 			MonoClass *klass = mono_object_class (ex);
 			if ((mono_runtime_unhandled_exception_policy_get () != MONO_UNHANDLED_POLICY_LEGACY) &&
 			    !is_threadabort_exception (klass)) {
-				mono_unhandled_exception_internal (&ex->object);
+				mono_unhandled_exception_internal (&ex->object, FALSE);
```

Comment:
Doesn't this mean that `suspend-on-unhandled` won't work for threads other than the main one?

```csharp
using System;
using System.Threading;

public class P
{
	public static void Main ()
	{
		var t = new Thread(() => {
				throw new Exception ("oops");
			});
		t.Start ();
		t.Join ();
	}
}
```

Reply:
Yes, that is correct. Looking at the existing logic for this feature, I thought that was how it was supposed to work.

 If neither of us is sure how it is supposed to work, I could do a git bisect to find the last version where this works to make sure I am duplicating the functionality in this case.

Reply:
I think you want to think about it as "if the debugger is attached, whenever it would get signalled that there's an unhandled exception, I should do this suspend loop". That is, find callers of this thing:
```
mini_get_dbg_callbacks ()->handle_exception (..., ..., NULL, NULL);
```
and do the suspend loop right before that callback.
(NULL, NULL means there's no catch clause)